File: 03-Age_in_Seconds/03.rb
#!/usr/bin/env ruby

require 'date'

birth = gets.chomp
begin
    parsed = Date.parse(birth)
rescue
    puts "Not a valid date! Please insert a date in the form YYYY/MM/DD or DD/MM/YYYY."
end

diff = (Date.today - parsed) * 60 * 60 * 24
puts diff.to_i

File: 03-Age_in_Seconds/03.py
#!/usr/bin/env python

from datetime import datetime as dt
import dateutil.parser as p


def main():
    try:
        date = p.parse(input("Your date of birth: "))
    except ValueError:
        return "That's not a valid date!"

    today = dt.today()

    print(int((today - date).total_seconds()))


if __name__ == "__main__":
    main()
